### PR TITLE
Always put null and undefined at the end of a union.

### PR DESCRIPTION
**Pull request by aomarks**

### CODE REVIEW

File: src/ts-ast.ts [hunk 1]
```diff
@@ -477,17 +477,30 @@ export class UnionType {
     // Functions too.
     const deduped = [];
     const names = new Set();
```
Comment (rictic) [inline src/ts-ast.ts:467]: BTW, sets are ordered in javascript, so you probably don't need both array and set representations here.

Also, better to be explicit about the element type of a set by initing it as `new Set<string>()`

**Review by rictic — APPROVED** [review]

**Comment by rictic** [comment]
Looks like travis is failing because it's trying to install polymer-analyzer @2.5.0 which doesn't exist?

File: src/ts-ast.ts (hunk 1 above)
Comment (aomarks) [reply src/ts-ast.ts:467]: `deduped` is in array of AST nodes, `names` is an array of strings, can't re-use. Updated types to be more clear.

**Comment by aomarks** [comment]
Yeah it's failing because it's based on https://github.com/Polymer/gen-typescript-declarations/pull/19 which needs a new Analyzer to be released.

File: src/ts-ast.ts (hunk 1 above)
Comment (rictic) [reply src/ts-ast.ts:467]: +1

**Pull request merged**